stea/stea_request.py

```diff
@@ -60,14 +62,27 @@ def add_ecl_profile(self, profile_id, key, first_year=None, last_year=None):
             end_date = case.end_date
             last_year = end_date.year
 
+        if multiplier is None:
+            multiplier = 1.0
+
         start_time = datetime.datetime(first_year, 1, 1)
         end_time = datetime.datetime(last_year + 1, 1,1)
         time_range = case.time_range(start=start_time, end=end_time, interval="1Y")
         ecl_unit = case.unit(key)
 
         unit = self.project.get_profile_unit(profile_id)
         mult = self.project.get_profile_mult(profile_id)
-        unit_conversion = self.units[unit][ecl_unit] * self.scale_factors[mult]
-        data = case.blocked_production(key, time_range) * unit_conversion
-        self.add_profile(profile_id, first_year, list(data))
+        if unit in self.units and ecl_unit in self.units[unit]:  
+          unitfactor = self.units[unit][ecl_unit]
+        else:
+          unitfactor = 1.0
+          sys.stdout.write('Default conversion between %s and %s to 1.\n' % (unit, ecl_unit))
+        unit_conversion = unitfactor * self.scale_factors[mult]
+        data = list(case.blocked_production(key, time_range) * unit_conversion)
+        if isinstance(multiplier, types.ListType):
```

Comment:
The ability to use a non-scalar multiplier seems overly complex to me - is it really necessary?

Reply:
it is used on real field testing.